File: package/main/src/Validate/string/length.ts
/**
 * String validation module for exact length check
 * Provides validation functionality for checking if a string has an exact length
 */

import type { ValidateReturnType } from "@/Validate/type";

/**
 * Creates a validator for checking if a string has an exact length
 * Note: The function name has an underscore suffix to avoid conflict with the `length` property
 * @param {number} length - The exact length that the string should have
 * @param {string} [message] - Custom error message for validation failure
 * @returns {ValidateReturnType<string>} - Validator for string length
 */
export const length_ = (
  length: number,
  message?: string,
): ValidateReturnType<string> => ({
  type: "string",
  message,
  validate: (value) => value.length === length,
});

File: package/umt_rust/src/validate/string/length.rs
//! String validation module for exact length check
//!
//! Provides validation functionality for checking if a string has an exact length.

use crate::validate::types::{ValidateReturnType, ValueType};

/// Creates a validator for checking if a string has an exact length
///
/// # Arguments
///
/// * `len` - The exact length that the string should have
/// * `message` - Custom error message for validation failure
///
/// # Returns
///
/// A `ValidateReturnType<String>` validator for string length
///
/// # Examples
///
/// ```
/// use umt_rust::validate::string::umt_length;
///
/// let validator = umt_length(3, None);
/// assert!((validator.validate)(&"abc".to_string()));
/// assert!(!(validator.validate)(&"ab".to_string()));
/// assert!(!(validator.validate)(&"abcd".to_string()));
/// ```
pub fn umt_length(len: usize, message: Option<&str>) -> ValidateReturnType<String> {
    ValidateReturnType::new(
        ValueType::String,
        move |value: &String| value.len() == len,
        message.map(String::from),
    )
}
